### ISSUE TITLE
Feature request: use an internal JSON beautifier

### ISSUE THREAD

**Issue by NTBBloodbath** [rest-nvim/rest.nvim#72]
I think I can do a pure Lua parser for beautifying JSON, in this way we can ditch jq for the JSON formatting and avoid having additional system-wide dependencies.

**Comment by Furkanzmc** [comment]
Hello.

I used to use my own HTTP client plugin. But I came across this one recently, and ditched my own in favour of yours. So, thanks for that! :)

Although I do consider dependencies as bad for plugins, I think `jq` is a good one. I would prefer being able to use `jq` and even have an option to pass the output through my own chain of commands than to use an internal parser that may or may not be as flexible.

I just noticed the issue and I thought I could give my 2 cents. Thanks for the good work!
